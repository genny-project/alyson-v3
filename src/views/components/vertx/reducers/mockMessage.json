{
  "data_type": "QBulkMessage",
  "messages": [
    {
      "items": [
        {
          "baseEntityAttributes": [
            {
              "baseEntityCode": "APP_PE07687867689768987967869768976876",
              "attributeCode": "PRI_GRP_PLACED_DATE",
              "readonly": false,
              "created": "2018-12-08T07:15:12",
              "valueDateTime": "2018-12-08T07:15:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE076876897698768967896787698678678",
              "attributeCode": "PRI_APPLICANT_USERNAME",
              "readonly": false,
              "created": "2018-12-08T07:02:38",
              "valueString": "jerry@hotmail.com",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0685688657856786578657865876578687",
              "attributeCode": "PRI_EDU_PROVIDER_CODE",
              "readonly": false,
              "created": "2018-12-08T07:02:48",
              "valueString": "CPY_OUTCOME_LIFE",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE67678867678768678768768786867786667",
              "attributeCode": "PRI_BEG_CODE",
              "readonly": false,
              "created": "2018-12-08T07:02:38",
              "valueString": "BEG_PE7686786786767886767867867776",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE089796876867986786776878786768687678",
              "attributeCode": "​​​​​PER_PINKIE_AT_YAHOOCOM​​​​​",
              "readonly": false,
              "created": "2018-12-08T07:11:34",
              "valueString": "green",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "STA_GREEN",
              "readonly": false,
              "created": "2018-12-08T07:11:34",
              "updated": "2018-12-08T07:11:34",
              "valueString": "[\n  \"​​​​​PER_JORDON_AT_GMAILCOM​​​​​\",\n  \"​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​\",\n  \"​​​​​PER_ROEL_AT_YAHOOCOM​​​​​\"\n]",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_GRP_APPLIED_DATE",
              "readonly": false,
              "created": "2018-12-08T07:02:41",
              "valueDateTime": "2018-12-08T07:02:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_GRP_SHORTLISTED_DATE",
              "readonly": false,
              "created": "2018-12-08T07:06:16",
              "valueDateTime": "2018-12-08T07:06:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_APPLICANT_CODE",
              "readonly": false,
              "created": "2018-12-08T07:02:38",
              "valueString": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_GRP_OFFERED_DATE",
              "readonly": false,
              "created": "2018-12-08T07:11:34",
              "valueDateTime": "2018-12-08T07:11:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "STA_​​​​​PER_ROEL_AT_YAHOOCOM​​​​​",
              "readonly": false,
              "created": "2018-12-08T07:11:34",
              "valueString": "green",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_COMPANY_CODE",
              "readonly": false,
              "created": "2018-12-08T07:02:41",
              "valueString": "CPY_PE2F8031F5BA6B4E2792D39BFD09641D77",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_APPLICANT_FULLNAME",
              "readonly": false,
              "created": "2018-12-08T07:02:38",
              "valueString": "Jerry Seinfeld",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "STA_​​​​​PER_JORDON_AT_GMAILCOM​​​​​",
              "readonly": false,
              "created": "2018-12-08T07:11:34",
              "valueString": "green",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_GRP_IN_PROGRESS_DATE",
              "readonly": false,
              "created": "2018-12-08T08:15:22",
              "valueDateTime": "2018-12-08T08:15:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_GRP_INTERVIEWS_DATE",
              "readonly": false,
              "created": "2018-12-08T07:07:48",
              "valueDateTime": "2018-12-08T07:07:00",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "attributeCode": "PRI_APPLICATION_CODE",
              "readonly": false,
              "created": "2018-12-08T07:02:38",
              "valueString": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            }
          ],
          "links": [
            {
              "link": {
                "attributeCode": "LNK_APP",
                "targetCode": "DOC_PE0D45F6A8BC3545D19C5DF09239CF9605",
                "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
                "linkValue": "INT_AGMT",
                "weight": 1
              },
              "created": "2018-12-08T07:11:34",
              "valueString": "INT_AGMT",
              "weight": 1
            },
            {
              "link": {
                "attributeCode": "LNK_APP",
                "targetCode": "CPY_OUTCOME_LIFE",
                "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
                "linkValue": "EDU_PROVIDER",
                "weight": 1
              },
              "created": "2018-12-08T07:02:41",
              "valueString": "EDU_PROVIDER",
              "weight": 1
            },
            {
              "link": {
                "attributeCode": "LNK_APP",
                "targetCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
                "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
                "linkValue": "APPLICANT",
                "weight": 1
              },
              "created": "2018-12-08T07:02:41",
              "valueString": "APPLICANT",
              "weight": 1
            }
          ],
          "code": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
          "index": 0,
          "created": "2018-12-08T07:02:38",
          "id": 7709,
          "name": "APPLICATION",
          "realm": "genny"
        }
      ],
      "linkCode": "LNK_BEG",
      "linkValue": "APPLICATION",
      "total": -1,
      "returnCount": 1,
      "data_type": "BaseEntity",
      "delete": false,
      "replace": false,
      "msg_type": "DATA_MSG"
    },
    {
      "items": [
        {
          "baseEntityAttributes": [
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_IMAGE_URL",
              "attributeName": "ImageUrl",
              "readonly": false,
              "created": "2018-12-08T07:01:17",
              "valueString": "https://s3.ap-southeast-2.amazonaws.com/channel40-images/jtaQozJT5HSpn3bX6P4EXSqQj3NDalkQ",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_LASTNAME",
              "attributeName": "LastName",
              "readonly": false,
              "created": "2018-12-08T07:01:17",
              "valueString": "Seinfeld",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_MOBILE",
              "attributeName": "Mobile",
              "readonly": false,
              "created": "2018-12-08T07:01:57",
              "valueString": "0423231333",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "LNK_INDUSTRY",
              "attributeName": "Industry",
              "readonly": false,
              "created": "2018-12-08T07:02:09",
              "valueString": "SEL_INDUSTRY_ENGINEERING",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_NAME",
              "attributeName": "Name",
              "readonly": false,
              "created": "2018-12-08T07:01:17",
              "valueString": "Jerry Seinfeld",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_USERNAME",
              "attributeName": "Username",
              "readonly": false,
              "created": "2018-12-08T07:01:17",
              "valueString": "jerry@hotmail.com",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "LNK_OCCUPATION",
              "attributeName": "Occupation",
              "readonly": false,
              "created": "2018-12-08T07:02:09",
              "updated": "2018-12-08T07:02:11",
              "valueString": "[\"SEL_OCCUPATION_DRAFTSPERSON\"]",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_FIRSTNAME",
              "attributeName": "FirstName",
              "readonly": false,
              "created": "2018-12-08T07:01:17",
              "valueString": "Jerry",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_IS_BUYER",
              "attributeName": "Buyer Role",
              "readonly": false,
              "created": "2018-12-08T07:01:36",
              "valueBoolean": false,
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "LNK_EDU_PROVIDER",
              "attributeName": "Edu Provider",
              "readonly": false,
              "created": "2018-12-08T07:02:01",
              "valueString": "CPY_OUTCOME_LIFE",
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            },
            {
              "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
              "attributeCode": "PRI_IS_SELLER",
              "attributeName": "Seller Role",
              "readonly": false,
              "created": "2018-12-08T07:01:36",
              "valueBoolean": true,
              "weight": 0,
              "inferred": false,
              "privacyFlag": false
            }
          ],
          "links": [],
          "code": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
          "index": 0,
          "created": "2018-12-08T07:01:17",
          "id": 7708,
          "name": "Jerry Seinfeld",
          "realm": "genny"
        }
      ],
      "total": -1,
      "returnCount": 1,
      "data_type": "BaseEntity",
      "delete": false,
      "replace": false,
      "msg_type": "DATA_MSG"
    }
  ]
}
