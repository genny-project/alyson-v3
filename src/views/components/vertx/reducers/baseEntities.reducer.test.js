import reducer from './baseEntities.reducer';
import mockMessage from './mockMessage.json';
import * as events from '../../../../utils/vertx/events';

const initialState = reducer( undefined, {});

const eventTypes = {
  DATA_MSG: 'data_type',
  CMD_MSG: 'cmd_type',
  EVT_MSG: 'event_type',
};

describe( 'baseEntities reducer', () => {
  describe( 'initial state', () => {
    it( 'should match snapshot', () => {
      expect( initialState ).toMatchSnapshot();
    });
  });

  let state = initialState;

  mockMessage.messages.forEach( message => {
    const { msg_type } = message;
    const eventType = eventTypes[msg_type];
    const event = message[eventType];
    const action = events[event]( message );

    state = reducer( state, action );
  });

  describe( 'base entity message', () => {
    const keys = Object.keys( state.data );

    it( 'should add data to the reducer', () => {
      expect( state ).toHaveProperty( 'data' );
      expect( keys.length ).toBeGreaterThan( 0 );
    });

    const firstApplication = state.data[keys[0]];
    const lastApplication = state.data[keys[keys.length - 1]];

    it( 'should store data in the correct format', () => {
      const expectedKeys = [
        'code',
        'created',
        'id',
        'index',
        'links',
        'name',
        'realm',
      ];

      expectedKeys.forEach( key => {
        expect( firstApplication ).toHaveProperty( key );
        expect( lastApplication ).toHaveProperty( key );
      });
    });
  });
});
