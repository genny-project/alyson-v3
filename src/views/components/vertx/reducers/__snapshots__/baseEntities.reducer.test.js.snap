// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseEntities reducer initial state should match snapshot 1`] = `
Object {
  "aliases": Object {},
  "attributes": Object {
    "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E": Object {
      "PRI_APPLICANT_CODE": Object {
        "attributeCode": "PRI_APPLICANT_CODE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:02:38",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "valueString": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "weight": 0,
      },
      "PRI_APPLICANT_FULLNAME": Object {
        "attributeCode": "PRI_APPLICANT_FULLNAME",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:02:38",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "Jerry Seinfeld",
        "valueString": "Jerry Seinfeld",
        "weight": 0,
      },
      "PRI_APPLICANT_USERNAME": Object {
        "attributeCode": "PRI_APPLICANT_USERNAME",
        "baseEntityCode": "APP_PE076876897698768967896787698678678",
        "created": "2018-12-08T07:02:38",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "jerry@hotmail.com",
        "valueString": "jerry@hotmail.com",
        "weight": 0,
      },
      "PRI_APPLICATION_CODE": Object {
        "attributeCode": "PRI_APPLICATION_CODE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:02:38",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "valueString": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "weight": 0,
      },
      "PRI_BEG_CODE": Object {
        "attributeCode": "PRI_BEG_CODE",
        "baseEntityCode": "APP_PE67678867678768678768768786867786667",
        "created": "2018-12-08T07:02:38",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "BEG_PE7686786786767886767867867776",
        "valueString": "BEG_PE7686786786767886767867867776",
        "weight": 0,
      },
      "PRI_COMPANY_CODE": Object {
        "attributeCode": "PRI_COMPANY_CODE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:02:41",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "CPY_PE2F8031F5BA6B4E2792D39BFD09641D77",
        "valueString": "CPY_PE2F8031F5BA6B4E2792D39BFD09641D77",
        "weight": 0,
      },
      "PRI_EDU_PROVIDER_CODE": Object {
        "attributeCode": "PRI_EDU_PROVIDER_CODE",
        "baseEntityCode": "APP_PE0685688657856786578657865876578687",
        "created": "2018-12-08T07:02:48",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "CPY_OUTCOME_LIFE",
        "valueString": "CPY_OUTCOME_LIFE",
        "weight": 0,
      },
      "PRI_GRP_APPLIED_DATE": Object {
        "attributeCode": "PRI_GRP_APPLIED_DATE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:02:41",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T07:02:00",
        "valueDateTime": "2018-12-08T07:02:00",
        "weight": 0,
      },
      "PRI_GRP_INTERVIEWS_DATE": Object {
        "attributeCode": "PRI_GRP_INTERVIEWS_DATE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:07:48",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T07:07:00",
        "valueDateTime": "2018-12-08T07:07:00",
        "weight": 0,
      },
      "PRI_GRP_IN_PROGRESS_DATE": Object {
        "attributeCode": "PRI_GRP_IN_PROGRESS_DATE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T08:15:22",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T08:15:00",
        "valueDateTime": "2018-12-08T08:15:00",
        "weight": 0,
      },
      "PRI_GRP_OFFERED_DATE": Object {
        "attributeCode": "PRI_GRP_OFFERED_DATE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:11:34",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T07:11:00",
        "valueDateTime": "2018-12-08T07:11:00",
        "weight": 0,
      },
      "PRI_GRP_PLACED_DATE": Object {
        "attributeCode": "PRI_GRP_PLACED_DATE",
        "baseEntityCode": "APP_PE07687867689768987967869768976876",
        "created": "2018-12-08T07:15:12",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T07:15:00",
        "valueDateTime": "2018-12-08T07:15:00",
        "weight": 0,
      },
      "PRI_GRP_SHORTLISTED_DATE": Object {
        "attributeCode": "PRI_GRP_SHORTLISTED_DATE",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:06:16",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "2018-12-08T07:06:00",
        "valueDateTime": "2018-12-08T07:06:00",
        "weight": 0,
      },
      "STA_GREEN": Object {
        "attributeCode": "STA_GREEN",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:11:34",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "updated": "2018-12-08T07:11:34",
        "value": Array [
          "​​​​​PER_JORDON_AT_GMAILCOM​​​​​",
          "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
          "​​​​​PER_ROEL_AT_YAHOOCOM​​​​​",
        ],
        "valueString": "[
  \\"​​​​​PER_JORDON_AT_GMAILCOM​​​​​\\",
  \\"​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​\\",
  \\"​​​​​PER_ROEL_AT_YAHOOCOM​​​​​\\"
]",
        "weight": 0,
      },
      "STA_​​​​​PER_JORDON_AT_GMAILCOM​​​​​": Object {
        "attributeCode": "STA_​​​​​PER_JORDON_AT_GMAILCOM​​​​​",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:11:34",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "green",
        "valueString": "green",
        "weight": 0,
      },
      "STA_​​​​​PER_ROEL_AT_YAHOOCOM​​​​​": Object {
        "attributeCode": "STA_​​​​​PER_ROEL_AT_YAHOOCOM​​​​​",
        "baseEntityCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
        "created": "2018-12-08T07:11:34",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "green",
        "valueString": "green",
        "weight": 0,
      },
      "​​​​​PER_PINKIE_AT_YAHOOCOM​​​​​": Object {
        "attributeCode": "​​​​​PER_PINKIE_AT_YAHOOCOM​​​​​",
        "baseEntityCode": "APP_PE089796876867986786776878786768687678",
        "created": "2018-12-08T07:11:34",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "green",
        "valueString": "green",
        "weight": 0,
      },
    },
    "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​": Object {
      "LNK_EDU_PROVIDER": Object {
        "attributeCode": "LNK_EDU_PROVIDER",
        "attributeName": "Edu Provider",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:02:01",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "CPY_OUTCOME_LIFE",
        "valueString": "CPY_OUTCOME_LIFE",
        "weight": 0,
      },
      "LNK_INDUSTRY": Object {
        "attributeCode": "LNK_INDUSTRY",
        "attributeName": "Industry",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:02:09",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "SEL_INDUSTRY_ENGINEERING",
        "valueString": "SEL_INDUSTRY_ENGINEERING",
        "weight": 0,
      },
      "LNK_OCCUPATION": Object {
        "attributeCode": "LNK_OCCUPATION",
        "attributeName": "Occupation",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:02:09",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "updated": "2018-12-08T07:02:11",
        "value": Array [
          "SEL_OCCUPATION_DRAFTSPERSON",
        ],
        "valueString": "[\\"SEL_OCCUPATION_DRAFTSPERSON\\"]",
        "weight": 0,
      },
      "PRI_FIRSTNAME": Object {
        "attributeCode": "PRI_FIRSTNAME",
        "attributeName": "FirstName",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:17",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "Jerry",
        "valueString": "Jerry",
        "weight": 0,
      },
      "PRI_IMAGE_URL": Object {
        "attributeCode": "PRI_IMAGE_URL",
        "attributeName": "ImageUrl",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:17",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "https://s3.ap-southeast-2.amazonaws.com/channel40-images/jtaQozJT5HSpn3bX6P4EXSqQj3NDalkQ",
        "valueString": "https://s3.ap-southeast-2.amazonaws.com/channel40-images/jtaQozJT5HSpn3bX6P4EXSqQj3NDalkQ",
        "weight": 0,
      },
      "PRI_IS_BUYER": Object {
        "attributeCode": "PRI_IS_BUYER",
        "attributeName": "Buyer Role",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:36",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": false,
        "valueBoolean": false,
        "weight": 0,
      },
      "PRI_IS_SELLER": Object {
        "attributeCode": "PRI_IS_SELLER",
        "attributeName": "Seller Role",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:36",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": true,
        "valueBoolean": true,
        "weight": 0,
      },
      "PRI_LASTNAME": Object {
        "attributeCode": "PRI_LASTNAME",
        "attributeName": "LastName",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:17",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "Seinfeld",
        "valueString": "Seinfeld",
        "weight": 0,
      },
      "PRI_MOBILE": Object {
        "attributeCode": "PRI_MOBILE",
        "attributeName": "Mobile",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:57",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "0423231333",
        "valueString": "0423231333",
        "weight": 0,
      },
      "PRI_NAME": Object {
        "attributeCode": "PRI_NAME",
        "attributeName": "Name",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:17",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "Jerry Seinfeld",
        "valueString": "Jerry Seinfeld",
        "weight": 0,
      },
      "PRI_USERNAME": Object {
        "attributeCode": "PRI_USERNAME",
        "attributeName": "Username",
        "baseEntityCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
        "created": "2018-12-08T07:01:17",
        "inferred": false,
        "privacyFlag": false,
        "readonly": false,
        "value": "jerry@hotmail.com",
        "valueString": "jerry@hotmail.com",
        "weight": 0,
      },
    },
  },
  "data": Object {
    "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E": Object {
      "code": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
      "created": "2018-12-08T07:02:38",
      "id": 7709,
      "index": 0,
      "links": Array [
        Object {
          "created": "2018-12-08T07:11:34",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "INT_AGMT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "DOC_PE0D45F6A8BC3545D19C5DF09239CF9605",
            "weight": 1,
          },
          "valueString": "INT_AGMT",
          "weight": 1,
        },
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "EDU_PROVIDER",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "CPY_OUTCOME_LIFE",
            "weight": 1,
          },
          "valueString": "EDU_PROVIDER",
          "weight": 1,
        },
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "APPLICANT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
            "weight": 1,
          },
          "valueString": "APPLICANT",
          "weight": 1,
        },
      ],
      "name": "APPLICATION",
      "realm": "genny",
    },
    "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​": Object {
      "code": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
      "created": "2018-12-08T07:01:17",
      "id": 7708,
      "index": 0,
      "links": Array [],
      "name": "Jerry Seinfeld",
      "realm": "genny",
    },
  },
  "definitions": Object {
    "data": Object {},
    "types": Object {},
  },
  "links": Object {
    "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E": Object {
      "APPLICANT": Array [
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "APPLICANT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
            "weight": 1,
          },
          "valueString": "APPLICANT",
          "weight": 1,
        },
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "APPLICANT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "​​​​​PER_JERRY_AT_HOTMAILCOM​​​​​",
            "weight": 1,
          },
          "valueString": "APPLICANT",
          "weight": 1,
        },
      ],
      "EDU_PROVIDER": Array [
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "EDU_PROVIDER",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "CPY_OUTCOME_LIFE",
            "weight": 1,
          },
          "valueString": "EDU_PROVIDER",
          "weight": 1,
        },
        Object {
          "created": "2018-12-08T07:02:41",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "EDU_PROVIDER",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "CPY_OUTCOME_LIFE",
            "weight": 1,
          },
          "valueString": "EDU_PROVIDER",
          "weight": 1,
        },
      ],
      "INT_AGMT": Array [
        Object {
          "created": "2018-12-08T07:11:34",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "INT_AGMT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "DOC_PE0D45F6A8BC3545D19C5DF09239CF9605",
            "weight": 1,
          },
          "valueString": "INT_AGMT",
          "weight": 1,
        },
        Object {
          "created": "2018-12-08T07:11:34",
          "link": Object {
            "attributeCode": "LNK_APP",
            "linkValue": "INT_AGMT",
            "sourceCode": "APP_PE0BA7075D03E94C9C8C42EEF6140FB73E",
            "targetCode": "DOC_PE0D45F6A8BC3545D19C5DF09239CF9605",
            "weight": 1,
          },
          "valueString": "INT_AGMT",
          "weight": 1,
        },
      ],
    },
  },
  "relationships": Object {},
}
`;
